### PR TITLE
KMeans init seed

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

**Comment by inteldimitrius** [comment]
Does pre-commit check failure make sense?

File: cpp/oneapi/dal/algo/kmeans_init/common.hpp [hunk 1]
```diff
@@ -74,6 +74,12 @@ constexpr bool is_valid_method_v = dal::detail::is_one_of_v<Method,
 template <typename Task>
 constexpr bool is_valid_task_v = dal::detail::is_one_of_v<Task, task::init>;
 
+template <typename M>
+constexpr bool is_not_default_dense = !std::is_same_v<M, method::dense>;
+
+template <typename M>
+using enable_if_not_default_dense = std::enable_if_t<is_not_default_dense<M>>;
+
```
Comment (Vika-F) [inline cpp/oneapi/dal/algo/kmeans_init/common.hpp:82]: Those redefinitions look like an overkill for me. They do not make the code simpler, but they require additional effort from reader to understand what is hidden underneath "enable_if_not_default_dense".

**Review by Vika-F — APPROVED** [review]
I'd prefer the code with less number of re-definitions, but I am Ok if the author has another opinion. In general the changes look good to me.

File: cpp/oneapi/dal/algo/kmeans_init/common.hpp (hunk 1 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/kmeans_init/common.hpp:82]: I believe it contains everything you need t know just in naming.

Comment (Vika-F) [reply cpp/oneapi/dal/algo/kmeans_init/common.hpp:82]: We do not need to agree on how particularly each line of code is implemented. We can have different sense of beauty, it is Ok.
In general the code looks good to me, I approve the changes.

**Review by napetrov — APPROVED** [review]

**Pull request merged**